multigeiger/multigeiger.ino

```diff
@@ -439,7 +439,16 @@ void loop()
   unsigned long time_difference;
   unsigned int HV_pulse_count;
   char sw[4];
-  unsigned long current_ms = millis();                      // to save multiple calls to millis()
+  unsigned long current_ms = millis();                           // to save multiple calls to millis()
+
+  // copy values from ISR
+  portENTER_CRITICAL(&mux_GMC_count);                            // enter critical section
+  GMC_counts = isr_GMC_counts;
+  isr_GMC_counts = 0;
+  count_timestamp = isr_count_timestamp;
+  GMC_counts_2send = isr_GMC_counts_2send;
+  count_timestamp_2send = isr_count_timestamp_2send;
+  portEXIT_CRITICAL(&mux_GMC_count);                             // leave critical section
```

Comment:
this needs to be done here so GMC_counts (and others) reflect fresh values.

later, we check for GMC_counts in multiple conditions and at the end of the loop, we set GMC_counts to 0 (which seems somehow superfluous?).